src/providers/krb5/krb5_child.c

```diff
@@ -40,15 +40,18 @@
 #include "util/sss_chain_id.h"
 #include "util/sss_ptr_hash.h"
 #include "src/util/util_errors.h"
-#include "responder/pam/pamsrv_passkey.h"
 #include "providers/backend.h"
 #include "providers/krb5/krb5_auth.h"
 #include "providers/krb5/krb5_utils.h"
 #include "krb5_plugin/idp/idp.h"
+#ifdef BUILD_PASSKEY
+#include "responder/pam/pamsrv_passkey.h"
 #include "krb5_plugin/passkey/passkey.h"
+#endif /* BUILD_PASSKEY */
 #include "sss_cli.h"
 
 #define SSSD_KRB5_CHANGEPW_PRINCIPAL "kadmin/changepw"
+#define SSSD_PASSKEY_QUESTION "passkey"
```

Comment:
I did this because it gets used in this code below
~~~
            } else if (strcmp(question_list[c], SSSD_PASSKEY_QUESTION) == 0) {
                return answer_passkey(ctx, kr, rctx);
            }                         
~~~
and I didn't want to `ifdef` this code so that `answer_passkey` can report a meaningful debug message with no passkey support is built. 

~~~
static krb5_error_code answer_passkey(krb5_context kctx,
                                      struct krb5_req *kr,
                                      krb5_responder_context rctx)
{   
#ifndef BUILD_PASSKEY
    DEBUG(SSSDBG_TRACE_FUNC, "Passkey auth not possible, SSSD built without passkey support!\n");
    return EINVAL;
#else
~~~

What do you suggest?

Reply:
I suggest to avoid double definition of `SSSD_PASSKEY_QUESTION` in case code is built with 'passkey' support.
Only define it locally if 'passkey' header isn't included.

Reply:
Fixed.

Reply:
Imo, putting it a couple lines above like
```
...
#include "krb5_plugin/passkey/passkey.h"
#else
#define SSSD_PASSKEY_QUESTION "passkey"
...
```
would be more readable but I don't insist.